crates/matrix_mentions/src/mention.rs

```diff
@@ -142,16 +149,45 @@ impl Mention {
     }
 }
 
+/// Determines if a uri can be parsed for a matrix id. Attempts to treat the uri in three
+/// ways when parsing:
+/// 1 - As a matrix uri
+/// 2 - As a matrix to uri
+/// 3 - As a custom uri
```

Comment:
I don't see any evidence that customisation of permalinks is part of the matrix spec, so yeah maybe it lies somewhere very specific. But I think it should still be easy to parameter,  not sure it lies in a non-default crate, as the platform code of the library is shipped with specific crate configuration. I think it could be a big drawback if people wanting to use custom permalink will have to maintain a fork of e.g. RTE's Swift Package.

Reply:
Addressed in [8417740](https://github.com/matrix-org/matrix-rich-text-editor/pull/719/commits/841774099585d1f22d6302d551a06b182c63303d)